hatch_pip_compile/lock.py

```diff
@@ -156,7 +158,9 @@ def get_file_content_hash(self) -> str:
         """
         Get hash of lock file
         """
-        return hashlib.sha256(self.lock_file.read_bytes()).hexdigest()
+        lockfile_contents = self.lock_file.read_bytes()
+        cross_platform_contents = lockfile_contents.replace(b"\r\n", b"\n")
+        return hashlib.sha256(cross_platform_contents).hexdigest()
```

Comment:
Dang, this was harder to find than it should've been. I had to go dig up an old windows laptop to get to the bottom of this 😅